src/hammer-vlsi/hammer_vlsi/hammer_build_systems.py

```diff
@@ -0,0 +1,244 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+#  hammer_build_systems.py
+#  Class containing all the methods to create VLSI flow build system infrastructure
+#
+#  See LICENSE for licence details.
+
+from .driver import HammerDriver
+
+import os
+import sys
+import textwrap
+from typing import List, Dict, Tuple, Callable
+
+def build_noop(driver: HammerDriver, append_error_func: Callable[[str], None]) -> dict:
+    """
+    Do nothing, just return the dependency graph.
+
+    :param driver: The HammerDriver
+    :return: The dependency graph
+    """
+    dependency_graph = driver.get_hierarchical_dependency_graph()
+    return dependency_graph
+
+
+def build_makefile(driver: HammerDriver, append_error_func: Callable[[str], None]) -> dict:
+    """
+    Build a Makefile include in the obj_dir called hammer.d. This is intended to be dynamically
+    created and included into a top-level Makefile.
+
+    The Makefile will contain targets for the following hammer actions, as well as any necessary
+    bridge actions (xyz-to-abc):
+        - pcb
+        - syn
+        - par
+        - drc
+        - lvs
+
+    For hierarchical flows, the syn, par, drc, and lvs actions will all be suffixed with the name
+    of the hierarchical modules (e.g. syn-Top, syn-SubModA, par-SubModA, etc.). The appropriate
+    dependencies and bridge actions are automatically generated from the hierarchy provided in the
+    Hammer IR.
+
+    Additionally, "redo" steps are created (e.g. redo-par for flat designs or redo-par-Top for
+    hierarchical), which allow the user to bypass the normal Makefile dependencies and force a
+    rerun of a particular task. This is useful when the user wants to change an input Hammer IR file
+    knowing it will not affect intermediate steps in the design.
+
+    An example use case for integrating this file into a top flow is provided below. Be sure to use
```

Comment:
Question, how does the snippet below include `$(OBJ_DIR)/hammer.d`?

Reply:
lol, it doesn't. I spent too much effort trying to make the example clean and forgot the most important line.

Reply:
Fixed.